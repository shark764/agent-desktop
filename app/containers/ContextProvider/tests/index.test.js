/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';

import { ContextProvider } from '../index';

// eslint-disable-next-line react/prefer-stateless-function
class TestApp extends React.Component {
  render() {
    if (this.context.toolbarMode) {
      return (
        <div>
          Toolbar
          {this.context.crmEnabled && <div>CRM Enabled</div>}
        </div>
      );
    } else {
      return (
        <div>
          Desktop
          {this.context.crmEnabled && <div>CRM Enabled</div>}
        </div>
      );
    }
  }
}

TestApp.contextTypes = {
  toolbarMode: PropTypes.bool,
  crmEnabled: PropTypes.bool,
};

describe('<ContextProvider />', () => {
  it('should set context flags correctly for desktop url', () => {
    const rendered = mount(
      <ContextProvider>
        <TestApp />
      </ContextProvider>
    );
    expect(rendered).toMatchSnapshot();
  });
  it('should set flags correctly when url for toolbar is hit', () => {
    Object.defineProperty(window.location, 'href', {
      writable: true,
      value: 'https://dev-tb2.cxengagelabs.net/',
    });
    const rendered = mount(
      <ContextProvider>
        <TestApp />
      </ContextProvider>
    );
    expect(rendered).toMatchSnapshot();
  });
});
