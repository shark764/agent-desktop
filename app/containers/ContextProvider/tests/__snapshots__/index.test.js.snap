// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContextProvider /> should set context flags correctly for desktop url 1`] = `
<ContextProvider>
  <TestApp>
    <div>
      Toolbar
    </div>
  </TestApp>
</ContextProvider>
`;

exports[`<ContextProvider /> should set flags correctly when url for toolbar is hit 1`] = `
<ContextProvider>
  <TestApp>
    <div>
      Toolbar
    </div>
  </TestApp>
</ContextProvider>
`;
